### ISSUE TITLE
Infinite loop when using Matcher.namedGroups

### ISSUE THREAD

**Issue by vprcic** [tony19/named-regexp#19]
It seems that if the pattern matches an empty string, an infinite loop occurs when using Matcher.namedGroups, even if the matched input isn't an empty string. Here's a code example:

```
Pattern pattern = Pattern.compile("(?<first>A)?(?<second>B)?");
Matcher matcher = pattern.matcher("AB");
if(matcher.matches()) {
    matcher.namedGroups();
}
```

**Comment by tony19** [comment]
Duplicate of #16. Fixed in #17. Not yet released.

**Comment by tony19** [comment]
Fixed in https://github.com/tony19/named-regexp/releases/tag/named-regexp-0.2.7
